motoman_driver/src/simple_message/motoman_motion_reply.cpp

```diff
@@ -157,6 +161,10 @@ std::string MotionReply::getSubcodeString(shared_int code)
     return "Waiting on ROS";
   case MotionReplySubcodes::NotReady::SKILLSEND:
     return "Waiting on SkillSend";
+  case MotionReplySubcodes::NotReady::PFL_ACTIVE:
+    return "PFL is active";
```

Comment:
If I remember correctly, PFL stands for power and force limiting and this error means that the driver can't be used when the robot is in PFL mode, correct? If so we might change this message to something more informative like:

```suggestion
    return "Power and force limiting (PFL) mode is active; disable PFL to execute trajectories with the ROS driver";
```

Reply:
There are some nuances to this I believe.

Perhaps @EricMarcil can clarify?

Reply:
ROS can operate the robot with the PFL function enabled.  This message is when the PFL has detected a force that is above the allowed threshold (like someone pushing on the robot).  So we do not want to 'disable' the PFL but simply 'reset' the PFL, which is normally done by pressing the amber button on the robot B-link.
After that, ROS will need to reinitialize the trajectory mode (ROS_CMD_START_TRAJ_MODE = 200121) before sending a new trajectory from the robot current position.

Reply:
So maybe a message like this would be better:

```suggestion
    return "Power and force limiting (PFL) has detected a force on the robot above the allowable threshold. Remove the obstruction and reset the PFL";
```

> After that, ROS will need to reinitialize the trajectory mode (ROS_CMD_START_TRAJ_MODE = 200121) before sending a new trajectory from the robot current position.

Is this something the user needs to do, or is it taken care of by the driver as a part of the trajectory execution action?

Reply:
New suggestion:

```suggestion
    return "Power and Force Limiting (PFL) was activated: "
      "please reset PFL on the robot, re-enable the driver (call the service) "
      "and send a new trajectory";
```

Reply:
> > After that, ROS will need to reinitialize the trajectory mode (ROS_CMD_START_TRAJ_MODE = 200121) before sending a new trajectory from the robot current position.
> 
> Is this something the user needs to do, or is it taken care of by the driver as a part of the trajectory execution action?

`motoman_driver` will try to enable trajectory mode upon reception of a new trajectory, but only if the driver has been "enabled" (by calling the `robot_enable` service) before.

It won't reset anything on the controller.

Reply:
Thanks for the input guys.

I've committed the new message for PFL.